src/utils/log_config.py

```diff
@@ -0,0 +1,178 @@
+        renderers - make the log message a string, json, dict, etc. with fancy styling
+        handlers -  print the rendered data to stdout, file, stream, etc.
+    """
+
+    # Obtain the module name that imported this log_config
+    caller = Path(inspect.stack()[-1].filename)
+    module_name = caller.name.split('.')[0]
+
+    # Gather the borealis configuration information
+    if not os.environ["BOREALISPATH"]:
+        raise ValueError("BOREALISPATH env variable not set")
+    if not os.environ['RADAR_CODE']:
+        raise ValueError('RADAR_CODE env variable not set')
+    config_path = f'{os.environ["BOREALISPATH"]}/config/' \
+                  f'{os.environ["RADAR_CODE"]}/'  \
+                  f'{os.environ["RADAR_CODE"]}_config.ini'
+    try:
+        with open(config_path, 'r') as config_data:
+            raw_config = json.load(config_data)
+    except IOError:
+        errmsg = f'Cannot open config file at {config_path}'
+        raise IOError(errmsg)
+
+    # If no override log level is set load the config log level
+    if log_level is None:
+        log_level = str(raw_config["log_level"])
+    if console is None:
+        console = raw_config["log_handlers"]["console"]
+    if logfile is None:
+        logfile = raw_config["log_handlers"]["logfile"]
+    if aggregator is None:
+        aggregator = raw_config["log_handlers"]["aggregator"]
+    # Set the log file and dir path. The time tag will be appended at the midnight
+    # roll over by the TimedRotatingLogHandler.
+    log_file = f"{raw_config['log_directory']}/{module_name}"
+
+    # Configure structlog here once for everything so that every log is uniformly formatted
+    structlog.configure(
+        # Processors are a list of functions that sequentially modify the event_dict (log message)
+        processors=[
+            structlog.stdlib.filter_by_level,  # Abort pipeline on log levels lower than threshold
+            # structlog.stdlib.add_logger_name,  # Add the name of the logger to event dict
+            structlog.stdlib.add_log_level,    # Add log level to event dict
+            structlog.processors.TimeStamper(fmt='iso', utc=True),  # Add ISO-8601 timestamp
+            structlog.processors.StackInfoRenderer(),  # Move "stack_info" in event_dict to "stack" and render
+            structlog.processors.UnicodeDecoder(),  # Decode byte strings to unicode strings
+            structlog.processors.CallsiteParameterAdder(  # Add items from the call enum
+                {
+                 structlog.processors.CallsiteParameter.FUNC_NAME,  # function name
+                 structlog.processors.CallsiteParameter.MODULE,  # module name
+                 structlog.processors.CallsiteParameter.PROCESS,  # process ID
+                 # structlog.processors.CallsiteParameter.THREAD,  # tread ID
+                 # structlog.processors.CallsiteParameter.FILENAME,  # file name
+                 # structlog.processors.CallsiteParameter.LINENO,  # line number
```

Comment:
Are these commented-out lines options we don't need?

Reply:
We won't want those in regular operation. It will only come up when we are wolf-fencing and debugging. These give the thread number, name of the file the log call was in, and which line in the file. This is just noise in normal operation.